ambari-server/src/main/java/org/apache/ambari/server/orm/dao/ServiceGroupDAO.java

```diff
@@ -43,15 +42,9 @@
   DaoUtils daoUtils;
 
   @RequiresSession
-  public ServiceGroupEntity findByPK(ServiceGroupEntityPK clusterServiceGroupEntityPK) {
-    return entityManagerProvider.get().find(ServiceGroupEntity.class, clusterServiceGroupEntityPK);
-  }
-
-  @RequiresSession
-  public ServiceGroupEntity findByClusterAndServiceGroupIds(Long clusterId, Long serviceGroupId) {
+  public ServiceGroupEntity findByPK(Long serviceGroupId) {
     TypedQuery<ServiceGroupEntity> query = entityManagerProvider.get()
-      .createNamedQuery("serviceGroupByClusterAndServiceGroupIds", ServiceGroupEntity.class);
-    query.setParameter("clusterId", clusterId);
+            .createNamedQuery("serviceGroupById", ServiceGroupEntity.class);
     query.setParameter("serviceGroupId", serviceGroupId);
```

Comment:
this is a very bad idea - you're not actually looking up the SG by a PK, so EclipseLink will always make a DB query. You should use the EntityManager to findByPK ...

Reply:
Got it. Fetching it like this now.

>> return entityManagerProvider.get().find(ServiceGroupEntity.class, serviceGroupId);

Similar change done for ServiceDAO as well.